contents/blog/ab-testing-guide-for-engineers.md

```diff
@@ -0,0 +1,252 @@
+---
+date: 2023-10-04
+title: "A software engineer's guide to A/B testing"
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+author: ['lior-neu-ner']
+featuredImage: ../images/blog/experiment-hog.jpg
+featuredImageType: full
+category: Product growth
+tags: 
+  - Guides
+  - Product metrics
+---
+
+A/B testing enables you to be confident that your product changes are having their intended effects. Rather than making decisions based on gut feeling or intuition, A/B testing provides empirical data to help you remove bias from decision making.
+
+This post is for software engineers who are just getting started with A/B testing. We'll teach you the basic concepts on how to run a successful test and supercharge your product growth!
+
+We'll cover:
+- [How A/B testing works](#how-ab-testing-works)
+- [What makes a good A/B test?](#what-makes-a-good-ab-test)
+- [How to implement a good A/B test](#how-to-implement-a-good-ab-test)
+- [Monitoring your A/B test](#monitoring-your-ab-test)
+- [Analyzing your results](#analyzing-your-results)
+- [Frequently asked questions](#frequently-asked-questions)
+- [Launch checklist](#launch-checklist)
+
+## How A/B testing works
+
+In A/B testing, you first decide on a goal metric and what change you'd like to see. For example, you may to increase conversion rate or decrease customer churn rate. 
+
+With your goal in mind and your code changes ready, you **randomly** split your users into two (or more) groups:
+
+- **The control group** – These users will experience your product as is – without any of your new changes.
+
+- **The test group** – These users will experience your product with your new changes.
+
+You then run your A/B test to gather data (usually a few days or weeks) and compare the differences in the goal metric between the different groups.
+
+![How A/B testing works](../images/blog/ab-testing-guide-for-engineers/ab-test-illustration.png)
+
+## What makes a good A/B test?
+
+There are five parts to a good A/B test:
+
+### 1. A clear, measurable goal
+
+Like "increase paid conversion rate", or "decrease churn rate". This should be a *single* metric that you're aiming to improve.
+
+Without a clear goal, it's unclear what is a successful outcome of test. For example, a bad goal is "increase user engagement", since it's unclear which metric will define success – is it daily active visitors, total number of page views, or time spent in the app?
```

Comment:
```suggestion
Without a clear goal, it's unclear what a successful outcome of the test is. For example, a bad goal is "increase user engagement", since it's unclear which metric will define success – is it daily active visitors, total number of page views, or time spent in the app?
```